src/ioc_scan/stix_extract.py

```diff
@@ -0,0 +1,143 @@
+- Hashes (SHA256, SHA1, MD5) of files, prioritizing by hash type.
+- Fully Qualified Domain Names (FQDNs), which can help identify associated domains.
+- URLs, which could represent potential threat sources or command and control servers.
+
+The script prints each observable type in a separate section with a clear title for easy reading.
+
+Usage:
+    stix-extract [<file>]
+
+Options:
+    -h --help   Show this screen.
+
+Arguments:
+    file        The path to the STIX xml file to parse. If not specified, reads from standard input.
+"""
+
+# Standard Python Libraries
+from collections import OrderedDict
+from io import TextIOWrapper
+import ipaddress
+import sys
+
+# Third-Party Libraries
+from docopt import docopt
+from stix.core import STIXPackage
+
+from ._version import __version__
+
+
+def extract_stix_info(stix_file):
+    """
+    Extract valuable information (IP addresses, hashes, FQDNs, and URLs) from a STIX file.
+
+    Prioritize hashes based on their type: SHA256 > SHA1 > MD5.
+
+    Args:
+        stix_file (str): path to the STIX file to parse
+
+    Returns:
+        tuple: a tuple containing four lists - one for IP addresses, one for hashes, one for FQDNs, and one for URLs.
+    """
+    # Load the STIX package from the XML file
+    try:
+        stix_package = STIXPackage.from_xml(stix_file)
+    except Exception as e:
+        sys.stderr.write(f"Error parsing STIX file: {e}")
+        raise e
+
+    # Initialize lists to store IP addresses, hashes, FQDNs, and URLs
+    ip_addresses = []
+    hashes = []
+    fqdns = []
+    urls = []
+
+    # Define hash type priority. Lower value means higher priority.
+    hash_priority = OrderedDict([("SHA256", 0), ("SHA1", 1), ("MD5", 2)])
+
+    # Iterate over each indicator in the STIX package
+    for indicator in stix_package.indicators:
+        for observable in indicator.observables:
+            object_type = observable.object_.properties._XSI_TYPE
+            if object_type == "AddressObjectType":
+                # Convert cybox.common.properties.String to str
+                ip_addresses.append(str(observable.object_.properties.address_value))
+            elif object_type == "FileObjectType":
+                hashes_dict = observable.object_.properties.hashes
+                if hashes_dict:
+                    best_hash = None
+                    best_priority = float("inf")
+                    for h in hashes_dict:
+                        if (
+                            h.type_.value in hash_priority
+                            and hash_priority[h.type_.value] < best_priority
+                        ):
+                            best_hash = str(h.simple_hash_value.value)  # Convert to str
+                            best_priority = hash_priority[h.type_.value]
+                    if best_hash is not None:
+                        hashes.append(best_hash)
+            elif object_type == "DomainNameObjectType":
```

Comment:
Minor but would you adjust this `elif` so that the `elif` checks are alphabetical? I think that would improve maintainability since we're just checking a bunch of strings.

Reply:
Not sure if you'd want to make those lists alphabetical or not considering we've done this before. LGTM tho